### PR TITLE
[Merged by Bors] - feat: version of the extreme value theorem without a non-emptiness assumption

### PR DESCRIPTION
**Pull request by grunweg**
Adapted and generalised from sphere-eversion; I'm just upstreaming it.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/Order/Compact.lean [hunk 1]
```diff
@@ -200,6 +200,17 @@ theorem IsCompact.exists_isMinOn [ClosedIicTopology α] {s : Set β} (hs : IsCom
   rcases (hs.image_of_continuousOn hf).exists_isLeast (ne_s.image f) with ⟨_, ⟨x, hxs, rfl⟩, hx⟩
   exact ⟨x, hxs, ball_image_iff.1 hx⟩
 
+/-- The **extreme value theorem**: a continuous function realizes its minimum on a compact set `s`.
+  This variant is for functions into a `NoMaxOrder`, without the assumption `s.Nonempty`. -/
+theorem IsCompact.exists_isMinOn' [ClosedIicTopology α] [NoMaxOrder α] {f : β → α}
```
Comment (urkud) [inline Mathlib/Topology/Algebra/Order/Compact.lean:205]: There is no `IsMinOn` here, so please rename. Also, I think that a better approach is to make the sphere-eversion project use [IsCompact.bddAbove_image](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Order/Compact.html#IsCompact.bddAbove_image) instead.

Comment (grunweg) [reply Mathlib/Topology/Algebra/Order/Compact.lean:205]: Good point, I have renamed the lemma.

Do think, however, that `IsCompact.bddAbove_image` or `bddBelow_image` do not suffice for sphere-eversion: these proofs need a lower minimum compatible with a lower bound on the image. (In the specific setting, `im(f)` is contained in the positive real numbers, and the proofs want a *positive* lower bound.)
It is possible to inline this lemma there. I have tried this, but imho it's *less* readable than having this lemma.

**Comment by grunweg** [comment]
awaiting-review

File: Mathlib/Topology/Algebra/Order/Compact.lean [hunk 2]
```diff
@@ -263,6 +263,17 @@ theorem IsCompact.exists_isMinOn [ClosedIicTopology α] {s : Set β} (hs : IsCom
   rcases (hs.image_of_continuousOn hf).exists_isLeast (ne_s.image f) with ⟨_, ⟨x, hxs, rfl⟩, hx⟩
   exact ⟨x, hxs, ball_image_iff.1 hx⟩
 
+/-- The **extreme value theorem**: a continuous function realizes its minimum on a compact set `s`.
+  This variant is for functions into a `NoMaxOrder`, without the assumption `s.Nonempty`. -/
```
Comment (fpvandoorn) [inline Mathlib/Topology/Algebra/Order/Compact.lean:267]: Please rewrite the docstring. This lemma states nothing about realizing its minimum. Maybe "if a continuous function lies strictly above `a` on a compact set, then it has a lower bound strictly above `a`."

bors d+

Comment (grunweg) [reply Mathlib/Topology/Algebra/Order/Compact.lean]: Good point and a much better docstring; thank you.

**Pull request closed without merging**